### PR TITLE
FAI-3432 - Support application tags & paths

### PR DESCRIPTION
**Pull request by willmarks**
Allows passing multiple tags and paths on a CD event that will be associated with the application being deployed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -273,6 +273,8 @@ There are two ways that arguments can be passed into the script. The first, is v
...
 | &#x2011;&#x2011;deploy_url            | The url of the deployment.                                                                                                                                                                                                            | --deploy   |
 | &#x2011;&#x2011;deploy_app_platform   | The compute platform that runs the application.                                                                                                                                                                                       | --deploy   |
+| &#x2011;&#x2011;deploy_app_tags       | A comma separated array of `key:value` application tags. (e.g. `key1:value1,key2:value2`)                                                                                                                                             | --deploy   |
+| &#x2011;&#x2011;deploy_app_paths      | A comma separated array of application paths. (e.g. `a/b/c,d/e/f`)                                                                                                                                                                    | --deploy   |
```
Comment (tovbinm) [inline README.md:277]:
```
A comma separated array of application slash separated paths. (e.g. `aws/us-east-1/eks-001,aws/us-west-2/eks-002`)
```

File: faros_event.sh [hunk 2]
```diff
@@ -884,6 +721,24 @@ function resolveDeployInput() {
...
+    fi
+    if ! [ -z "$deploy_app_paths" ]; then
+        paths_regex="^[^,]+(,[^,]+)*$"
+        if [[ "$deploy_app_paths" =~ $paths_regex ]]; then
+            IFS=',' read -ra deploy_app_path_arr <<< "$deploy_app_paths"
+        else
+            err "deploy_app_paths could not be parsed"
+            fail
+        fi
+    fi
```
Comment (tovbinm) [inline faros_event.sh:741]: Can we instead allow sending application tags & paths as strings and parse in `/events` API and error out of fails?

Comment (tovbinm) [reply faros_event.sh:741]: We did something similar before already, where we allowed passing either an array or a string for some argument.

**Review by tovbinm — COMMENTED** [review]
see comments

File: faros_event.sh (hunk 2 above)
Comment (willmarks) [reply faros_event.sh:741]: Sounds good! The change is [here](https://github.com/faros-ai/clio/pull/471)

**Review by tovbinm — APPROVED** [review]

File: test/spec/faros_event_spec.sh [hunk 3]
```diff
@@ -37,6 +37,8 @@ Describe 'faros_event.sh'
           --deploy_source "<deploy_source>" \
           --deploy_url "<deploy_url>" \
           --deploy_app_platform "<deploy_app_platform>" \
+          --deploy_app_tags "<key1>:<value1>,<key2>:<value2>" \
```
Comment (tovbinm) [inline test/spec/faros_event_spec.sh:40]: What's the default value for tags and paths if nothing is passed in ? Null?

Comment (willmarks) [reply test/spec/faros_event_spec.sh:40]: The keys are not added to the event body.

**Pull request merged**